istioctl/cmd/istioctl/inject.go

```diff
@@ -267,6 +268,14 @@ istioctl kube-inject -f deployment.yaml -o deployment-injected.yaml --injectConf
 					return err
 				}
 			} else {
+				// ISTIOCTL_USE_BUILTIN_DEFAULTS is used to have legacy behaviour.
+				if !getBoolEnv("ISTIOCTL_USE_BUILTIN_DEFAULTS", false) {
+					return errors.New("one of injectConfigFile or injectConfigMapName is required\n" +
```

Comment:
injectConfigMap should be the default ( since we default to fetch the mesh config as well ). 

So the message should be only if we fail to get the config map (and likely the mesh config) - something like "failed to retrieve mesh config from k8s, please specify injectConfigFile")